gto/registry.py

```diff
@@ -225,13 +232,21 @@ def promote(
             )
         return promotion
 
-    def check_ref(self, ref: str):
-        "Find out what was registered/promoted in this ref"
+    def _check_ref(self, ref, state):
+        if ref.startswith("refs/tags/"):
+            ref = ref[len("refs/tags/") :]
+        if ref.startswith("refs/heads/"):
+            ref = self.repo.commit(ref).hexsha
         return {
-            "version": self.version_manager.check_ref(ref, self.get_state()),
-            "stage": self.stage_manager.check_ref(ref, self.get_state()),
+            "version": self.version_manager.check_ref(ref, state),
+            "stage": self.stage_manager.check_ref(ref, state),
         }
 
+    def check_ref(self, ref: str):
```

Comment:
@amritghimire, this is not ideal, but if you create `GitRegistry`, you can use this function body as an example on how to cache state. Looking for better solutions.

Reply:
This could work. A similar approach for get_stages will also be great. We do checkout to a commit for each get_stages call. But if get_stages use the head when creating GitRegistry, having ref there as well would be better.

Reply:
Why do you call get_stages each time? What stages user sees in MR should depend on config HEAD only. So it doesn't matter what is there in all other commits.

Reply:
E.g. you should always call `$ gto stages` when HEAD of default branch gets updated.

Reply:
We need to parse everything for each commit. Because: 
1. We parse incrementally and not all at once
2. Some commits might be removed, in this case we don't want to reparse, we want to recalc.

In perfect world we wouldn't need to create `GitRegistry` and manage it. Things will be cached for us on your side, not sure how hard is this to do though.

Reply:
The easiest workaround is probably providing `git_registry` param, i.e. if not `None` then use it instead of recreating each time.